hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactSerializationTest.java

```diff
@@ -430,6 +439,125 @@ public void testReflectiveSerializer_withFieldsWithOtherSerializationMechanisms_
         assertTrue(data.isCompact());
     }
 
+    @Test
+    public void testCompactWriterThrowsExceptionWhileWritingWhenSchemaDoesNotHaveThatField() {
+        thrown.expect(HazelcastSerializationException.class);
+        thrown.expectMessage("Failed to serialize");
+        thrown.expectCause(new RootCauseMatcher(HazelcastSerializationException.class, "Invalid field name"));
```

Comment:
Personally, I don't like this style of asserting exceptions are thrown. What if any line other than the last one throws this? I believe the test will pass again, despite us expecting it to throw in the last line. I would suggest using assertThatThrownBy for the last line. The same applies to other test as well

Reply:
Ok makes sense, I was not used to it. Let me try to switch to it.